### DISCUSSION TITLE
Cookies with chrono DateTime

### DISCUSSION THREAD

**Discussion by DreamTexX** [trillium-rs/trillium#634] [Q&A] [answered]
How do i convert a chrono DateTime to an OffsetDateTime, so i can use it to set the expire attribute on a cookie?

**Comment by jbr** [comment, marked as answer]
Hi, thanks for using trillium!

It's been a bit since I've used the `chrono` crate but looking at the docs suggests you should be able to convert between them by way of a unix timestamp:

```rust
let offset_datetime = time::OffsetDateTime::from_unix_timestamp(chrono_datetime.timestamp()).unwrap();
```

For other use cases you might want to use sub second precision, but there really isn't any reason to be that precise for cookie expiry. We also don't need to retain the timezone offset because expiry doesn't happen in any particular timezone.

Trillium currently exposes types from the [`cookie`](https://docs.rs/cookie/latest/cookie/) crate, so their choice of time library is also exposed.

**Reply by DreamTexX** [reply]
Hey, I finally came up with this solution too. Would it be possible to add a feature for chrono to the cookie crate, which would make it possible to use "chrono" types instead of forcing the "time" crate?

**Reply by jbr** [reply]
In theory yes, but it looks like that's been discussed and PRed and closed unmerged: https://github.com/rwf2/cookie-rs/pull/213

Even better in my opinion would be a opt-in feature on `time` or `chrono` that provided From implementation for the other crate.

However, all of that is outside of the scope of trillium and is more of a general rust ecosystem problem
